modelskill/comparison/_comparer_plotter.py

```diff
@@ -79,32 +82,66 @@ def timeseries(
         backend : str, optional
             use "plotly" (interactive) or "matplotlib" backend,
             by default "matplotlib"
+        style: list of str, optional
+            containing line styles of the model results. Cannot be passed together with color input.
+            by default None
+        color: list of str, optional
+            containing colors of the model results.
+            If len(colors) == num_models + 1, the first color will be used for the observations.
+            Cannot be passed together with style input.
+            by default None
         **kwargs
             other keyword arguments to fig.update_layout (plotly backend)
 
         Returns
         -------
         matplotlib.axes.Axes or plotly.graph_objects.Figure
         """
+
         from ._comparison import MOD_COLORS
 
         cmp = self.comparer
 
         if title is None:
             title = cmp.name
 
+        if color is not None and style is not None:
+            raise ValueError(
+                "It is not possible to pass both the color argument and the style argument. Choose one."
+            )
+
+        # Color for observations:
+        obs_color = cmp.data[cmp._obs_name].attrs["color"]
+
+        # if color is None and style is None:  # Use default values for colors
+        #     from ._comparison import MOD_COLORS
+
+        #     color = MOD_COLORS[: cmp.n_models]
+
+        color, style = _check_kwarg_and_convert_to_list(color, style, cmp.n_models)
+
+        if color is not None and len(color) > cmp.n_models:
+            # If more than n_models colors is given, the first color is used for the observations
+            obs_color = color[0]
+            color = color[1:]
+
         if backend == "matplotlib":
             fig, ax = _get_fig_ax(ax, figsize)
             for j in range(cmp.n_models):
                 key = cmp.mod_names[j]
                 mod = cmp.raw_mod_data[key]._values_as_series
-                mod.plot(ax=ax, color=MOD_COLORS[j])
+                if style is not None:
+                    mod.plot(ax=ax, style=style[j])
```

Comment:
I suggest to add a check (outside the loop) that the length of the style list is the same as the number of models (or models+1 obs, see comment below).